classes/mender-install.bbclass

```diff
@@ -1,23 +1,24 @@
-#Add meta information to the created image
-inherit mender-image-buildinfo
-IMAGE_PREPROCESS_COMMAND += "buildinfo_mender;"
+PREFERRED_VERSION_go_cross = "1.6%"
 
-EXTRA_IMAGEDEPENDS += "u-boot u-boot-fw-utils"
-PREFERRED_VERSION_u-boot = "v2015.10%"
-PREFERRED_VERSION_u-boot-fw-utils = "v2015.10%"
-PREFERRED_VERSION_go_cross = "1.5%"
-
-#Add support for systemd
+# Add support for systemd
+#
+# Note that a very common error with systemd is that the system hangs during the
+# boot process while looking for devices. This is almost always because the
+# kernel feature CONFIG_FHANDLE is not enabled.
 DISTRO_FEATURES_append = " systemd"
 VIRTUAL-RUNTIME_init_manager = "systemd"
 DISTRO_FEATURES_BACKFILL_CONSIDERED = "sysvinit"
 VIRTUAL-RUNTIME_initscripts = ""
 
+IMAGE_INSTALL_append = " mender ca-certificates"
 
-#Make sure we are creating sdimg with all needed partitioning.
-IMAGE_CLASSES += "sdimg"
-IMAGE_FSTYPES += "sdimg"
-
-IMAGE_INSTALL_append = " u-boot-fw-utils mender ca-certificates"
+# The storage device that holds the device partitions.
+MENDER_STORAGE_DEVICE ?= "/dev/mmcblk0"
 
+# The base name of the devices that hold individual partitions.
+# This is often MENDER_STORAGE_DEVICE + "p".
+MENDER_STORAGE_DEVICE_BASE ?= "/dev/mmcblk0p"
 
+# The numbers of the two rootfs partitions in the A/B partition layout.
+MENDER_PARTITION_A_NUMBER ?= "2"
```

Comment:
I know that this might be not needed at the moment (or at all), but since we are changing this anyway maybe we should just add `MENDER_ROOTFS_PART_A = "dev/mmcblk0p2"` and `MENDER_ROOTFS_PART_B = "dev/mmcblk0p3"`. The reason is that what will happen if we will decide that active partition will be internal flash and inactive will be SD card. The option of using full partition string will work in all cases.